code/modules/spells/spell_types/soultap.dm

```diff
@@ -18,13 +23,20 @@
 	action_icon_state = "soultap"
 
 /obj/effect/proc_holder/spell/self/tap/cast(list/targets, mob/living/user = usr)
+	if(HAS_TRAIT(user, TRAIT_NO_SOUL))
+		to_chat(user, span_warning("You have no soul to tap into!"))
+		return
+
 	to_chat(user, span_danger("Your body feels drained and there is a burning pain in your chest."))
 	user.maxHealth -= HEALTH_LOST_PER_SOUL_TAP
 	user.health = min(user.health, user.maxHealth)
 	if(user.maxHealth <= 0)
 		to_chat(user, span_userdanger("Your weakened soul is completely consumed by the tap!"))
 		return
```

Comment:
This also used to remove your soul, back in the day. Not that it mattered, you were out of health

Reply:
i'll re-add it, sure